lib/hexapdf/type/image.rb

```diff
@@ -254,6 +257,48 @@ def png_chunk(type, data = '')
         [data.length].pack("N") << type << data << [Zlib.crc32(data, Zlib.crc32(type))].pack("N")
       end
 
+      # Writes the image as TIFF to the given IO stream.
+      def write_tiff(io, info)
+        stream = stream_source
+
+        # header
+        io << ['49492A00'].pack('H*')
+
+        # offset to IFD
+        pad_size = stream.length % 2
+        io << [stream.length + 8 + pad_size].pack('L')
+
+        # data
+        io << Filter.string_from_source(stream)
+
+        # pad to word boundary
+        io << [0].pack('C') * pad_size
+
+        decode_params = [self[:DecodeParms]].flatten[0]
+        compression = if decode_params[:K] > 0 # G3-2D
+                        3
+                      elsif decode_params[:K] < 0 # G4-2D
+                        4
+                      else # G3-1D
+                        2
+                      end
+
+        # IFD entries
+        ifd_entries = [
+            [0x0100, 3, 1, info.width], # width
+            [0x0101, 3, 1, info.height], # height
+            [0x0102, 3, 1, info.bits_per_component], # bits per sample
+            [0x0103, 3, 1, compression], # compression
+            [0x0106, 3, 1, decode_params[:BlackIs1] === false ? 1 : 0], # photometric interpretation
```

Comment:
We can just switch on `decode_params[:BlackIs1]` here since it is a boolean.

Reply:
`decode_params[:BlackIs1]` could be `nil`

Even though the standard default value is `false`, macOS Preview app seems to default to true.

I have encountered a few cases where `[0x0106, 3, 1, decode_params[:BlackIs1] ? 0 : 1]` generates invereted files (while Preview app renders the images correctly)